core/Tracker/PageUrl.php

```diff
@@ -335,6 +338,44 @@ public static function reconstructNormalizedUrl($url, $prefixId)
         return $fullUrl;
     }
 
+    /**
+     * Returns if the given host is also configured as https in page urls of given site
+     *
+     * @param $idSite
+     * @param $host
+     * @return false|mixed
+     * @throws \Exception
+     */
+    public static function shouldUseHttpsHost($idSite, $host)
+    {
+        $cache = Cache::getTransientCache();
+
+        $cacheKeySiteUrls = sprintf('siteurls-%s', $idSite);
```

Comment:
This is minor and not strictly necessary but we could use `CacheId::siteAware` here for consistency.